src/ceph-disk/ceph_disk/main.py

```diff
@@ -3756,6 +3756,25 @@ def main_activate(args):
             )
             osd_data = get_mount_point(cluster, osd_id)
 
+            args.cluster = cluster
+
+            # Check if encrypted block device in journal
+
+            block_dmcrypt_path = os.path.join(osd_data, 'block_dmcrypt')
+
+            if args.dmcrypt and os.path.exists(block_dmcrypt_path):
+                partition = DevicePartition.factory(
+                    path=None,
+                    dev=os.path.join(osd_data, 'block_dmcrypt'),
+                    args=args)
+
+                partition.osd_dm_key = get_dmcrypt_key(
```

Comment:
i see. `partition.set_dev(dev)` points both the rawdev and dev to `block_dmcrypt_path`. so maybe we can just put
```py
partition.rawdev = args.path
```
before 
```py
partition.map()
```

what do you think?